drivers/sensor/bmi08x/bmi08x_gyro.c

```diff
@@ -0,0 +1,525 @@
+	for (i = 0; i < ARRAY_SIZE(bmi08x->gyr_sample); i++) {
+		bmi08x->gyr_sample[i] = sys_le16_to_cpu(bmi08x->gyr_sample[i]);
+	}
+
+	return 0;
+}
+
+static void bmi08x_to_fixed_point(int16_t raw_val, uint16_t scale, struct sensor_value *val)
+{
+	int32_t converted_val;
+
+	/*
+	 * maximum converted value we can get is: max(raw_val) * max(scale)
+	 *	max(raw_val) = +/- 2^15
+	 *	max(scale) = 4785
+	 *	max(converted_val) = 156794880 which is less than 2^31
+	 */
+	converted_val = raw_val * scale;
+	val->val1 = converted_val / 1000000;
+	val->val2 = converted_val % 1000000;
+}
+
+static void bmi08x_channel_convert(enum sensor_channel chan, uint16_t scale, uint16_t *raw_xyz,
+				   struct sensor_value *val)
+{
+	int i;
+	uint8_t ofs_start, ofs_stop;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+		ofs_start = ofs_stop = 0U;
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		ofs_start = ofs_stop = 1U;
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		ofs_start = ofs_stop = 2U;
+		break;
+	default:
+		ofs_start = 0U;
+		ofs_stop = 2U;
+		break;
+	}
+
+	for (i = ofs_start; i <= ofs_stop; i++, val++) {
+		bmi08x_to_fixed_point(raw_xyz[i], scale, val);
+	}
+}
+
+static inline void bmi08x_gyr_channel_get(const struct device *dev, enum sensor_channel chan,
+					  struct sensor_value *val)
+{
+	struct bmi08x_gyro_data *bmi08x = dev->data;
+
+	bmi08x_channel_convert(chan, bmi08x->scale, bmi08x->gyr_sample, val);
+}
+
+static int bmi08x_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct bmi08x_gyro_data *bmi08x = dev->data;
+
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(dev, &state);
+	if (state != PM_DEVICE_STATE_ACTIVE) {
+		return -EBUSY;
+	}
+#endif
+
+	switch ((int16_t)chan) {
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		bmi08x_gyr_channel_get(dev, chan, val);
+		return 0;
+	default:
+		LOG_ERR("Channel not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int bmi08x_gyro_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	uint8_t reg_val;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		reg_val = BMI08X_GYRO_PM_NORMAL;
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		reg_val = BMI08X_GYRO_PM_SUSPEND;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	if (bmi08x_gyro_byte_write(dev, BMI08X_REG_GYRO_LPM1, reg_val) < 0) {
+		LOG_ERR("Failed to set power mode");
+		return -EIO;
+	}
+	k_msleep(BMI08X_GYRO_POWER_MODE_CONFIG_DELAY);
+
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static const struct sensor_driver_api bmi08x_api = {
+	.attr_set = bmi08x_attr_set,
+#ifdef CONFIG_BMI08X_GYRO_TRIGGER
+	.trigger_set = bmi08x_trigger_set_gyr,
+#endif
+	.sample_fetch = bmi08x_sample_fetch,
+	.channel_get = bmi08x_channel_get,
+};
+
+int bmi08x_gyro_init(const struct device *dev)
+{
+	const struct bmi08x_gyro_config *config = dev->config(dev);
```

Comment:
const struct bmi08x_gyro_config *config = dev->config;

dev->config is not a function call